### ISSUE TITLE
Error when a variant has no `productId`

### ISSUE THREAD

**Issue by Sam-Black-0013** [craftcms/commerce#1003]
### Description
I have a super table field (by verbb), with a variant field in it.
If i try to select a variant, i get this error:
`PHP Fatal Error 'yii\base\ErrorException' with message 'Method craft\commerce\elements\Variant::__toString() must not throw an exception, caught yii\base\InvalidConfigException: Invalid product ID: 1225'`

I also tried to use it outside the super table field, same problem,

This error used to occur in the discount page where the id of a deleted product  would throw an exception if the discount have a variant of that deleted product.

And the product with id 1225 does not exist anymore

The variants and elements records in the db did not get  deleted when the product got deleted
### Additional info

- Craft version: 3.2.10
- PHP version: 7.1
- Commerce: 2.1.12.1
- Super table: 2.2.1

**Comment by lukeholder** [comment]
The variant elements should have been deleted when the product was.

Not sure how you got in this state, did you make any custom product/variant imports or have any module code manipulating the variants?

You can delete all variants without a productId with

```
DELETE e
FROM elements e
inner join commerce_variants cv on `e`.`id` = `cv`.`id`
WHERE `cv`.`productId` is null
```

**Comment by lukeholder** [comment]
Can you give any steps to reproduce?

**Comment by Sam-Black-0013** [comment]
I have been developing this site locally for months now and I deleted a lot of testing products

I never had this issue because I never used the field before

I did use feed me when I started developing the site (back in november 2018)

It was set to add new products and I used it just once that time. I never used it to update its data nor did i use anything else to manipulate the product data.

So i am not really sure how I got to this state.
